zstd_stream.go

```diff
@@ -168,17 +168,13 @@ func (w *Writer) Write(p []byte) (int, error) {
 		srcData = w.srcBuffer
 	}
 
-	var srcPtr *byte // Do not point anywhere, if src is empty
-	if len(srcData) > 0 {
-		srcPtr = &srcData[0]
-	}
-
+	// &srcData[0] is safe: it is p or w.srcBuffer but only if len() > 0 checked above
```

Comment:
I would still do some defensive coding do an if statement like the others (but that's just personal opinion), here we assume anyone making change to the `Write` method will read that comment (i.e: if I remove `if len(p) ==0` condition above, I might not read this comment)

Reply:
This is a good point. This is also reasonably complex logic to figure out that this is safe: it involves following some complex operations above. In this case, we could do another "early return": `if len(srcData) == 0 { return 0, nil }`. I will make this change.